File: src/main/java/com/github/masx200/biliClient/model/relation/Relation.java
package com.github.masx200.biliClient.model.relation;

import com.github.masx200.biliClient.BiliResult;
import com.alibaba.fastjson.JSONObject;
import lombok.Data;

import java.util.List;

/**
 * 描述：
 *
 * @author lpc lpc@hll520.cn
 * @version 1.0  2021-02-08-16:32
 * @since 2021-02-08-16:32
 */
@Data
public class Relation {
    /**
     * 总数
     */
    private long total;

    /**
     * 乐观锁版本
     */
    private long re_version;

    /**
     * 关注者列表
     */
    private List<Follower> items;

    /**
     * 工根据响应结果构建
     *
     * @param result 响应结果
     * @return rl
     */
    public static Relation build(BiliResult result) {
        Relation relation = result.toData(Relation.class);
        List<Follower> list = JSONObject.parseObject(result.getData().toString())
                .getJSONArray("list").toJavaList(Follower.class);
        relation.setItems(list);
        return relation;
    }
}

File: src/main/java/com/github/masx200/biliClient/model/relation/Relation.kt
package com.github.masx200.biliClient.model.relation

import com.github.masx200.biliClient.BiliResult
import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonNames

//import com.alibaba.fastjson.JSONObject;
/**
 * 描述：
 *
 * @author lpc lpc@hll520.cn
 * @version 1.0 2021-02-08-16:32
 * @since 2021-02-08-16:32
 */
@Serializable
data //@com.alibaba.fastjson2.annotation.JSONCompiled
class Relation(
    /**
     * 总数
     */
    val total: Long = 0,

    /**
     * 乐观锁版本
     */
    val re_version: Long = 0,

    /**
     * 关注者列表
     */
    var items: MutableList<Follower?>? = mutableListOf()
) {
    companion object {
        /**
         * 工根据响应结果构建
         *
         * @param result 响应结果
         * @return rl
         */
        @JvmStatic
        fun build(result: BiliResult): Relation {
            val relation: Relation = result.toData<Relation>()
            val list: MutableList<Follower?>? = result.toData<DataFollowersRoot>()
                .list
            relation.items=(list)
            return relation
        }
    }
}

@Serializable
@OptIn(ExperimentalSerializationApi::class)
data class DataFollowersRoot(
    val list: MutableList<Follower?>?,
    @JsonNames("re_version")
    val reVersion: Long,
    val total: Long,
)
